ssz-rs/src/ser.rs

```diff
@@ -71,14 +78,15 @@ pub fn serialize_composite_from_components(
 
     // SAFETY: `fixed_lengths_sum` fits in `u32` if the total size check holds
     let mut running_length = fixed_lengths_sum as u32;
-    debug_assert_eq!(fixed.len(), variable_lengths.len());
     for (part_opt, variable_length) in fixed.iter_mut().zip(variable_lengths) {
         if let Some(part) = part_opt {
             buffer.append(part);
         } else {
             // SAFETY: `variable_length` fits in `u32` if the total size check holds
             let bytes_written = running_length.serialize(buffer)?;
-            debug_assert_eq!(bytes_written, BYTES_PER_LENGTH_OFFSET);
+            if bytes_written != BYTES_PER_LENGTH_OFFSET {
+                return Err(SerializeError::UnexpectedSize(bytes_written, BYTES_PER_LENGTH_OFFSET))
+            }
```

Comment:
right -- the only way to trigger this error is if the code above changes; otherwise we know statically that `bytes_written` will be `BYTES_PER_LENGTH_OFFSET` and hence why I went w/ the `debug_assert` here

I kind of lean towards just leaving this one part alone